component/metrics/globalrefmap.go

```diff
@@ -44,7 +44,7 @@ func newGlobalRefMap() *GlobalRefMap {
 }
 
 // GetOrAddLink is called by a remote_write endpoint component to add mapping and get back the global id.
-func (g *GlobalRefMap) GetOrAddLink(componentID string, localRefID uint64, l labels.Labels) uint64 {
```

Comment:
Note for later: 

GlobalRefMap is going to be the cause of a lot of bottlenecking because of how much locking it does. The biggest problem is how this is using a Lock and not a RWLock, but even if that weren't true, there's still many concurrent write operations that will be bottlenecked by the other one. 

I would suggest to start thinking about how to make the cache and global IDs as lock free as possible.

Reply:
I was going to hold off on making significant changes to the global ref mapping until it either a. becomes a problem or b. we do submodules. Whatever we decide to do with submodules the global ref mapping will probably have to change a fair amount (not to mention other code).

Reply:
It doesn't have to be changed right now, but I just wanted to point it out as a concern because it introduces much more lock contention across multiple scrape jobs than Prometheus Agent or the static Grafana Agent does.